rust/oak/src/storage/mod.rs

```diff
@@ -34,31 +34,32 @@ pub struct Storage {
     read_channel: crate::ReadHandle,
 }
 
-impl Default for Storage {
+impl Storage {
     /// Create a default `Storage` instance assuming the standard port names
     /// (`"storage_in"`, `"storage_out"`) for pre-defined channels for storage
     /// communication.
-    fn default() -> Storage {
+    pub fn default() -> Option<Storage> {
         Storage::new("storage_in", "storage_out")
     }
-}
 
-impl Storage {
     /// Create a `Storage` instance using the given port names for pre-defined
     /// channels for storage communication.
-    pub fn new(in_port_name: &str, out_port_name: &str) -> Storage {
+    pub fn new(in_port_name: &str, out_port_name: &str) -> Option<Storage> {
         let read_handle = ReadHandle {
             handle: crate::channel_find(in_port_name),
         };
         let write_handle = WriteHandle {
             handle: crate::channel_find(out_port_name),
         };
+        if read_handle.handle == 0 || write_handle.handle == 0 {
```

Comment:
Could you add a constant definition for those `0`s? Even nicer, if `channel_find` returned an `Option` or `Result` instead, this would read much more nicely I think.

Reply:
Added an `INVALID_HANDLE` constant for now (shifting to `Option<Handle>` can happen separately).